lib/modules/manager/rust/extract.ts

```diff
@@ -0,0 +1,26 @@
+import { regEx } from '../../../util/regex';
+import { GithubReleasesDatasource } from '../../datasource/github-releases';
+import { regexMatchAll } from '../custom/regex/utils';
+import type { PackageDependency, PackageFileContent } from '../types';
+
+const channelRegex = regEx(
+  /channel\s*=\s*"(?<currentValue>\d+\.\d+(\.\d+)?)"/,
+  'g',
+);
```

Comment:
My preference: always capture the channel= line, no matter what the value is. Then let versioning take care of whether it's something we can upgrade or not. At least users will know that we found it, and it's not upgradeable